rest_service.go

```diff
@@ -54,10 +81,32 @@ func (s *Service) App() *gimlet.APIApp {
 	app.AddRoute("/process/{id}/signal/{signal}").Version(1).Patch().Handler(s.signalProcess)
 	app.AddRoute("/close").Version(1).Delete().Handler(s.closeManager)
 	app.AddRoute("/process/{id}/logs").Version(1).Get().Handler(s.getLogs)
+	app.AddRoute("/configure-cache").Version(1).Post().Handler(s.configureCache)
+	app.AddRoute("/download-mongodb").Version(1).Post().Handler(s.downloadMongoDB)
+
+	go s.backgroundPrune()
 
 	return app
 }
 
+func (s *Service) backgroundPrune() {
+	s.cacheMutex.RLock()
+	timer := time.NewTimer(s.cacheOpts.PruneDelay)
+	s.cacheMutex.RUnlock()
+
+	for {
+		<-timer.C
+		s.cacheMutex.RLock()
```

Comment:
Do we expect that `cacheOpts` will change after they're set?

Reply:
Yes. At the moment, the client can only change cacheOpts by using the `/configure-cache` route, so `backgroundPrune()` is already running by the time the client has a chance to set their own options.